include/flatbuffers/flatbuffers.h

```diff
@@ -464,12 +464,10 @@ class vector_downward {
  public:
   explicit vector_downward(size_t initial_size,
                            const simple_allocator &allocator)
-    : reserved_(initial_size),
+    : reserved_(initial_size & ~(sizeof(largest_scalar_t) - 1)),
```

Comment:
maybe round it up? right now, if you pass `4` you will get `0`, which will cause problems.